eth2/lmd_ghost/src/reduced_tree.rs

```diff
@@ -182,13 +183,15 @@ impl ReducedTreeSsz {
         }
     }
 
-    pub fn to_reduced_tree<T, E>(
+    pub fn into_reduced_tree<T, E>(
         self,
         store: Arc<T>,
         block_root_tree: Arc<BlockRootTree>,
     ) -> Result<ReducedTree<T, E>> {
         if self.node_hashes.len() != self.nodes.len() {
-            Error::InvalidReducedTreeSsz("node_hashes and nodes should have equal length".into());
```

Comment:
The error here was not returned. I think the intended behaviour is to return the Error. Fixed it in PR but I want to make sure that is correct.

Reply:
Oops my bad. Yup it's supposed to return the error. 
Thanks :)